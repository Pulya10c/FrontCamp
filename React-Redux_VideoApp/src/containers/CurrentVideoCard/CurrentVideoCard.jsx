import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import {
  filmPosterPathSelector,
  filmTitleSelector,
  filmRatingSelector,
  filmReleaseDateSelector,
  filmOverviewSelector
} from "selectors/currentVideoSelectors";

const mapStateToProps = state => ({
  posterPath: filmPosterPathSelector(state),
  title: filmTitleSelector(state),
  rating: filmRatingSelector(state),
  releaseDate: filmReleaseDateSelector(state),
  overview: filmOverviewSelector(state)
});

const CurrentVideoCard = ({
  posterPath,
  title,
  rating,
  releaseDate,
  overview
}) => {
  return (
    <>
      <div>
        <img src={posterPath} />
      </div>
      <div>
        <h2>{title} </h2>
        <p>
          Rating: {rating} Release date: {releaseDate}
        </p>
        <p>{overview}</p>
      </div>
    </>
  );
};

CurrentVideoCard.propTypes = {};

export default connect(mapStateToProps, null)(CurrentVideoCard);
