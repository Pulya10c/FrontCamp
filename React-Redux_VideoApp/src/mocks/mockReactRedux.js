const createMockMapStateToProps = mapStateToProps => {
  if (!mapStateToProps) {
    return mapStateToProps;
  }

  return (state, ownProps) => {
    return {
      props: mapStateToProps(state, ownProps)
    };
  };
};

const getDispatchFunctionFormShorthandObject = actionsObj => dispatch => {
  return Object.keys(actionsObj).reduce((props, actionKey) => {
    return {
      ...props,
      [actionKey]: function() {
        return dispatch(actionsObj[actionKey].apply(this, arguments));
      }
    };
  }, {});
};

const createMockMapDispatchToProps = mapDispatchToProps => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps;
  }

  const mappingFunction =
    typeof mapDispatchToProps === 'function'
      ? mapDispatchToProps
      : getDispatchFunctionFormShorthandObject(mapDispatchToProps);

  return (dispatch, ownProps) => {
    dispatch = dispatch || jest.fn(action => action);

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    };
  };
};

const createMockMergeProps = mergeProps => {
  if (!mergeProps) {
    return mergeProps;
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: jest.fn(action => action) };

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    };
  };
};

export default {
  connect: (
    mapStateToProps,
    mapDispatchToProps,
    mergeProps
  ) => ConnectedComponent => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    ConnectedComponent
  })
};
