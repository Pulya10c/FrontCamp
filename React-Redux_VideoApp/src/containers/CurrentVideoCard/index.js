import CurrentVideoCard from './CurrentVideoCard';

export default CurrentVideoCard;
