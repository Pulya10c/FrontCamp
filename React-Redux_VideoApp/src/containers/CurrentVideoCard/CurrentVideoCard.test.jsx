import React from "react";
import { shallow } from "enzyme";
import mockReactRedux from "mocks/mockReactRedux";
import {
  filmPosterPathSelector,
  filmTitleSelector,
  filmRatingSelector,
  filmReleaseDateSelector,
  filmOverviewSelector
} from "selectors/currentVideoSelectors";
import CurrentVideoCard from "./CurrentVideoCard";

jest.mock("react-redux", () => mockReactRedux);

const mockFilmPosterPath = "mockFilmPosterPath";
const mockFilmTitle = "mockFilmTitle";
const mockRating = 5;
const mockReleaseDate = "mockDate";
const mockOverview = "mockOverview";

jest.mock("selectors/currentVideoSelectors", () => ({
  filmPosterPathSelector: jest.fn(() => mockFilmPosterPath),
  filmTitleSelector: jest.fn(() => mockFilmTitle),
  filmRatingSelector: jest.fn(() => mockRating),
  filmReleaseDateSelector: jest.fn(() => mockReleaseDate),
  filmOverviewSelector: jest.fn(() => mockOverview)
}));

describe("Component CurrentVideoCard", () => {
  const { mapStateToProps, ConnectedComponent } = CurrentVideoCard;

  it("should render correctly", () => {
    const { props } = mapStateToProps();
    expect(shallow(<ConnectedComponent {...props} />)).toMatchSnapshot();
  });

  it("should pass correct props to component from mapStateToProps", () => {
    const { props } = mapStateToProps();
    expect(props).toEqual({
      posterPath: mockFilmPosterPath,
      title: mockFilmTitle,
      rating: mockRating,
      releaseDate: mockReleaseDate,
      overview: mockOverview
    });
  });
});
