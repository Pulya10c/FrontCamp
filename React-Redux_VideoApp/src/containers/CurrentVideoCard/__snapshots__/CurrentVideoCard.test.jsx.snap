// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CurrentVideoCard should render correctly 1`] = `
<styled.div>
  <div>
    <img
      src="mockFilmPosterPath"
    />
  </div>
  <div>
    <h2>
      mockFilmTitle
       
    </h2>
    <p>
      Rating: 
      5
       Release date: 
      mockDate
    </p>
    <p>
      mockOverview
    </p>
  </div>
</styled.div>
`;
